packages/auth/amplify_auth_cognito_android/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AtomicResult.kt

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.amazonaws.amplify.amplify_auth_cognito
+
+import io.flutter.Log
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * Thread-safe [NativeAuthPluginBindings.Result] wrapper which prevents multiple replies and
+ * automatically posts results to the main thread.
+ */
+class AtomicResult<T>(
```

Comment:
Should the one in core eventually be replaced with this one?

Reply:
This wraps the generated bindings' result which is different than a generic Flutter result. So they're incompatible and we'll need both, unfortunately, unless at some point we migrate all the plugins to a single pigeon definition in core which could be pretty cool.